In [5]:
a1 = "Alisson"
a2 = "Ariane"
print a1 + " e " + a2

Alisson e Ariane


In [6]:
pet_halibut = "Why should I be tarred with the epithet "loony" merely because I have a pet halibut?"


SyntaxError: invalid syntax (<ipython-input-6-19d6efb1c7d7>, line 1)

In [7]:
pet_halibut = 'Why should I be tarred with the epithet "loony" merely because I have a pet halibut?'


In [8]:
salesman = '"I think you\'re an encyclopaedia salesman"'
print salesman

"I think you're an encyclopaedia salesman"


# Quiz: Conserte a aspa
A linha de código no quiz seguinte causará um erro de sintaxe (SyntaxError), devido ao uso errado das aspas. Primeiro, execute-a com o Executar Teste para visualizar a mensagem de erro. Depois, resolva o problema de forma que a aspa seja corretamente atribuída à variável messiah.


In [10]:
# TODO: Fix this string!
messiah = 'He\'s not the Messiah, he\'s a very naughty boy!'
print messiah

He's not the Messiah, he's a very naughty boy!


In [11]:
a = "Alisson disse que queria um copo d\'aguá"
print a

Alisson disse que queria um copo d'aguá


QUESTION 2 OF 3

Nós já vimos como o tipo dos objetos afetará a maneira como os operadores trabalham com eles. Qual será a saída deste código?

coconut_count = "34"
mango_count = "15"
tropical_fruit_count = coconut_count + mango_count
print(tropical_fruit_count)

In [12]:
coconut_count = "34"
mango_count = "15"
tropical_fruit_count = coconut_count + mango_count
print(tropical_fruit_count)

3415


A resposta correta é 3415 e tropical_fruit_count é uma string. Isso ocorre porque mesmo que os caracteres dentro das cadeias coconut_count e mango_count sejam números, os valores da variável coconut_count e mango_count são strings e são tratados como strings quando se trata do operador + que está sendo aplicado.
Devido aos seus parênteses, isso o tornou uma string


In [13]:
type(coconut_count)

str

In [15]:
type(mango_count)

str

Quiz: Escreva uma mensagem de log do servidor
Neste quiz de programação, você vai utilizar o que aprendeu sobre strings para escrever uma mensagem de log de um servidor.

Você receberá um dado de exemplo com o usuário, o horário da visita e o site acessado. Você deve usar as variáveis fornecidas e as técnicas aprendidas para imprimir uma mensagem de log como esta (com o usuário, url e registro de tempo substituídos pelos valores apropriados das variáveis):

Yogesh accessed the site http://petshop.com/pets/reptiles/pythons at 16:20.

Utilize o botão Executar Teste para ver os resultados enquanto você trabalha na codificação parte por parte.

In [39]:
# 1 
username = "Alisson"
timestamp = "08:16"
url = "http://www.webmotors.com.br"
print(username + " accessed the site " + url + " at " + timestamp)

Alisson accessed the site http://www.webmotors.com.br at 08:16


In [40]:
# Conforme a pagina da Udacity

username = "Kinari"
timestamp = "04:50"
url = "http://petshop.com/pets/mammals/cats"
print(username + " accessed the site " + url + " at " + timestamp + ".")

# TODO: print a log message incorporating the strings above.
# The message should be use the same format as this one:
# "Yogesh accessed the site http://petshop.com/pets/reptiles/pythons at 16:20."

username = "Gi"
timestamp = "22:00"
url = "http://codecademy.com.br/html-&-css"
print(username + " accessed the site " + url + " at " + timestamp+ ".")

Kinari accessed the site http://petshop.com/pets/mammals/cats at 04:50.
Gi accessed the site http://codecademy.com.br/html-&-css at 22:00.


# Funções embutidas

In [45]:
udacity_lenght = len("Udacity")
print udacity_lenght

7
